services/cloudfront.py

```diff
@@ -2,40 +2,137 @@
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+        if (
+            "Logging" in distribution["DistributionConfig"]
+            and distribution["DistributionConfig"]["Logging"]["Enabled"] == True
+        ):
+            compliant_resources.append(distribution["ARN"])
+        else:
+            non_compliant_resources.append(distribution["ARN"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_associated_with_waf():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]["Items"]
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        if "WebACLId" in distribution["DistributionConfig"] and distribution["DistributionConfig"]["WebACLId"] != "":
+            compliant_resources.append(distribution["ARN"])
+        else:
+            non_compliant_resources.append(distribution["ARN"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_default_root_object_configured():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]["Items"]
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        if distribution["DistributionConfig"]["DefaultRootObject"] != "":
+            compliant_resources.append(distribution["ARN"])
+        else:
+            non_compliant_resources.append(distribution["ARN"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_no_deprecated_ssl_protocols():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]["Items"]
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        for origin in distribution["DistributionConfig"]["Origins"]["Items"]:
+            if (
+                "CustomOriginConfig" in origin
+                and "SSLv3" in origin["CustomOriginConfig"]["OriginSslProtocols"]["Items"]
+            ):
```

Comment:
Origin과 HTTP로만 통신하는 경우 적용되지 않습니다.
```suggestion
    for distribution in distributions:
        for origin in distribution["Origins"]["Items"]:
            if (
                "CustomOriginConfig" in origin
                and origin["CustomOriginConfig"]["OriginProtocolPolicy"] in ["https-only", "match-viewer"]
                and "SSLv3" in origin["CustomOriginConfig"]["OriginSslProtocols"]["Items"]
            ):
```